src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReader.java

```diff
@@ -90,8 +90,8 @@ public void initialize(InputSplit inputSplit, TaskAttemptContext taskAttemptCont
     try {
       AuthenticatorCredentials credentials = SalesforceConnectionUtil.getAuthenticatorCredentials(conf);
       bulkConnection = new BulkConnection(Authenticator.createConnectorConfig(credentials));
-      InputStream queryResponseStream = waitForBatchResults(bulkConnection, jobId, batchId);
-      setupParser(queryResponseStream);
+      String[] resultIds = waitForBatchResults(bulkConnection, jobId, batchId);
```

Comment:
Could resultIds be null?

Reply:
It won't be null. https://github.com/forcedotcom/wsc/blob/7fcabe96ce4071d0f3693c05636b829f113b9496/src/main/java/com/sforce/async/QueryResultList.java#L53